app/controllers/bars_controller.rb

```diff
@@ -4,7 +4,12 @@ def index
   end
 
   def new
-    @bar = Bar.new
+    if user_signed_in?
```

Comment:
To avoid having to do this in each method, check out the `before_action` helper that you can place at the top of your controller!

http://guides.rubyonrails.org/action_controller_overview.html#filters